flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/table/MongoDBConnectorDeserializationSchema.java

```diff
@@ -129,29 +129,56 @@ public void deserialize(SourceRecord record, Collector<RowData> out) throws Exce
                 extractBsonDocument(value, valueSchema, MongoDBEnvelope.DOCUMENT_KEY_FIELD);
         BsonDocument fullDocument =
                 extractBsonDocument(value, valueSchema, MongoDBEnvelope.FULL_DOCUMENT_FIELD);
-
+        BsonDocument fullDocumentBeforeChange =
+                extractBsonDocument(
+                        value, valueSchema, MongoDBEnvelope.FULL_DOCUMENT_BEFORE_CHANGE_FIELD);
```

Comment:
IIUC, we need to extract this document when `full.document.pre.post.image` config is enabled. We can avoid the unnecessary cost for lower mongo version if we know the config.

Reply:
Fixed by passing `scan.full.changelog` config to the deserializer.